index.js

```diff
@@ -500,6 +509,38 @@ module.exports = class NoiseSecretStream extends Duplex {
     cb(null)
   }
 
+  async _boxMessage (buffer) {
+    if (!this._boxSecret) await new Promise(resolve => this.once('handshake', resolve))
```

Comment:
dont rely on events for this, i think we have a promise in the constructor you can await, otherwise lets add

Reply:
actually see comment below about moving this part out